py_scripts/parser.py

```diff
@@ -0,0 +1,203 @@
+"""
+useful urls = 
+            
+            'https://stackoverflow.com/questions/2869564/xml-filtering-with-python'
+            'https://stackoverflow.com/questions/1912434/how-do-i-parse-xml-in-python'
+
+@themkdemiiir stated:
+
+    The project involves creating a new parser to effectively 
+    manage and integrate the comprehensive rare diseases dataset
+    from Orphanet, which is available on their Orphanet Scientific Knowledge
+    Files website. This data is rich in content, including rare disease alignments
+    with various medical classifications like SNOMED CT, ICD-10, ICD-11, OMIM, UMLS, MeSH, MedDRA, and GARD. 
+    In addition, it provides detailed information on classifications of 
+    rare diseases, their linearisation, associated genes, clinical signs and symptoms,
+    functional consequences, epidemiology, and natural history.
+    We aim to develop a parser to handle these diverse data types and
+    bring them into a unified format that fits our system's needs.
+    This will significantly enhance our capacity to utilize the Orphanet
+    dataset for advanced analysis and application in rare disease studies.
+
+
+We are going to use the ElementTree lib to parse the XML file.
+The data 'en_product3_181.xml' is a 4.1MB for rare neurological diseases.
+
+"""
+
+import pandas as pd
+import xml.etree.ElementTree as ET
+import os
+from tqdm import tqdm
+
+
+raw_data_path = '../orphanet_data/orphanet_xml'
+directory = '../orphanet_data/orphanet_csv'
+
+
+
+class Disease:
+    
+    def __init__(self, classification_id, disorder_type, disorder_id,
+                 orpha_code, name, expert_link, meta_data):
+        
+        self.classification_id = classification_id
+        self.disorder_id = disorder_id
+        self.disorder_type = disorder_type
+        self.orpha_code = orpha_code
+        self.name = name
+        self.expert_link = expert_link
+        self.meta_data = meta_data
+        
+    def to_dict(self):
+        
+        return {
+            
+            'ClassificationID': self.classification_id,
+            'DisorderID': self.disorder_id,
+            'DisorderType' : self.disorder_type,
+            'OrphaCode': self.orpha_code,
+            'Name': self.name,
+            'ExpertLink': self.expert_link,
+            'MetaData': self.meta_data
+            
+        }
```

Comment:
If a class is required for a certain task, it is recommended to use data classes.

Reply:
For current main.py, I have only used functions. Please review the latest pr.